lightning/src/util/ser_macros.rs

```diff
@@ -1065,6 +1065,10 @@ macro_rules! impl_writeable_tlv_based_enum {
 /// when [`MaybeReadable`] is practical instead of just [`Readable`] as it provides an upgrade path for
 /// new variants to be added which are simply ignored by existing clients.
 ///
+/// Note that variants written as tuple variants will not support downgrading no matter the
+/// `$tuple_variant_id` used. Any variants written as a normal variant will support downgrading by
+/// having [`MaybeReadable`] return `None` for odd `$variant_id` variants not understood.
```

Comment:
Is it accurate to phrase this more strongly as "new tuple variants using this macro MUST use even types"? Seems a bit clearer.